tools/metadata_generation/swapi_metadata_generator.py

```diff
@@ -0,0 +1,97 @@
+"""SWAPI metadata generator."""
+
+import json
+
+import pandas as pd
+
+
+def get_global_attributes(sheet):
+    """Get the global attributes from a metadata spreadsheet.
+
+    Parameters
+    ----------
+    sheet : pandas.DataFrame
+        A sheet from an Excel file represented as a pandas DataFrame.
+
+    Returns
+    -------
+    dict
+        A dictionary of the global attributes, where the keys are the attribute names
+        and the values are the attribute descriptions.
+    """
+    sheet = sheet.set_index("NAME")
+    # subset by global_att field type, then only keep the CATDESC column
+    sheet = sheet[sheet["FIELD_TYPE"] == "global_att"][["CATDESC"]]
+    return sheet.to_dict()["CATDESC"]
+
+
+def get_file_attributes(sheet, global_attrs=None):
+    """Get the file attributes from a metadata spreadsheet.
+
+    Parameters
+    ----------
+    sheet : pandas.DataFrame
+        A sheet from an Excel file represented as a pandas DataFrame.
+
+    Returns
+    -------
+    dict
+        A dictionary of the file attributes, where the keys are the attribute names
+        and the values are the attribute descriptions.
+    """
+    # Add the global attributes for this specific sheet
+    file_attrs = {"global_attrs": get_global_attributes(sheet)}
+    # Add the global attributes from the global_attrs sheet (if any)
+    file_attrs["global_attrs"].update(global_attrs or {})
+
+    # Move onto variable and coordinate metadata
+    # Get everything that isn't a global attribute and drop that column
+    variables = sheet[sheet["FIELD_TYPE"] != "global_att"].drop(columns=["FIELD_TYPE"])
+    variables = variables.set_index("NAME")
+
+    # Iterate through the rows of the sheet, each row being a variable or coordinate
+    # Then drop any NaN values and convert the row to a dictionary of key/values like:
+    # {"CATDESC": "description", "VAR_TYPE": "support_data", ...}
+    for row, data in variables.iterrows():
+        file_attrs[row] = data.dropna().to_dict()
```

Comment:
How does dropping all NaN values affect ISTP compliance for the required attributes?

Reply:
Another great question! This is something we should talk about next week. Where do these checks occur. Do we want it to occur in this tool, upstream in the spreadsheet itself, or downstream in some other validator (possibly just relying on cdflib?). I was definitely going for the "just convert" approach here and was assuming validation is done elsewhere.